Automattic/woocommerce-payments
# fix: error message rendering on tokenized PRBs

Fixes #

#### Changes proposed in this Pull Request









While testing the tokenized PRB, I noticed that sometimes the error thrown by the Store API might not be rendered correctly.
Fixing.

#### Testing instructions





- On your server, change [this line](https://github.com/Automattic/woocommerce-payments-server/blob/b8960b4ccb5a3352958e32f1b8b9ab9a5b172e4b/server/wp-content/rest-api-plugins/endpoints/wcpay/class-base-controller.php#L342-L342) to be `if ( true || ...`
- Disable ECE with the following snippet:
```php
add_filter( 'pre_option__wcpay_feature_stripe_ece', function () {
    return '0'; 
}, 100 );
```
- Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
- Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
- Go to a product page
- Attempt to make a payment
- Close the GooglePay dialog
- The message should be correctly rendered on the page

Before:
<img width="845" alt="Screenshot 2024-07-30 at 11 53 34 AM" src="https://github.com/user-attachments/assets/1837a938-cb71-4c14-a58e-a969808aa3f8">

After:
<img width="891" alt="Screenshot 2024-07-30 at 11 53 11 AM" src="https://github.com/user-attachments/assets/109ece6e-fa08-42b3-baeb-3b819ef969a8">


-------------------


**Post merge**
